controller/controller_server/utils.py

```diff
@@ -151,23 +159,46 @@ def validate_create_snapshot_request(request):
 def validate_delete_snapshot_request(request):
     logger.debug("validating delete snapshot request")
     if not request.snapshot_id:
-        raise ValidationException(messages.name_should_not_be_empty_message)
+        raise ValidationException(messages.snapshot_id_should_not_be_empty_message)
     logger.debug("validating secrets")
     if request.secrets:
         validate_secret(request.secrets)
     logger.debug("request validation finished.")
 
 
+def validate_validate_volume_capabilities_request(request):
+    logger.debug("validating validate_volume_capabilities request")
+
+    logger.debug("validating volume id")
+    volume_id = request.volume_id
+    if not volume_id:
+        raise ValidationException(messages.volume_id_should_not_be_empty_message)
+    if config.PARAMETERS_OBJECT_ID_DELIMITER not in volume_id:
+        raise ObjectIdError(config.VOLUME_TYPE_NAME, volume_id)
+
+    logger.debug("validating parameters")
+    if request.parameters:
+        _validate_create_volume_parameters(request.parameters)
+
+    validate_csi_volume_capabilities(request.volume_capabilities)
+
+    validate_secret(request.secrets)
+
+
+def validate_volume_context_match_volume(volume_context, volume):
+    logger.debug("validate volume_context is matching volume")
+    context_from_existing_volume = _get_context_from_volume(volume)
+
+    if volume_context != context_from_existing_volume:
+        raise ValidationException(messages.volume_context_not_match_volume_message)
```

Comment:
would we be able to tell from our logs which part of the context is mismatched?

Reply:
done